pkgs/by-name/vm/vmctl/package.nix

```diff
@@ -0,0 +1,44 @@
+{
+  lib,
+  stdenv,
+  bash,
+  shellcheck,
+  fetchFromGitHub,
+}:
+
+stdenv.mkDerivation {
+  pname = "vmctl";
+  version = "1";
+
+  src = fetchFromGitHub {
+    owner = "SamsungDS";
+    repo = "vmctl";
+    rev = "5b6b7084b8cba06b474c0e020df965237f2c832c";
+    sha256 = "sha256-yDgaY2RJXBjWkMSQb4JaJzMGLFzowfOGixSRzzv2iIk=";
+  };
+
+  nativeBuildInputs = [
+    bash
+    shellcheck
+  ];
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp vmctl $out/bin
+    cp -r cmd $out/bin
+    cp -r common $out/bin
+    cp -r contrib $out/bin
+    cp -r lib $out/bin
```

Comment:
It seems weird to me that all of these are going to `$out/bin`. I suppose it is because the entrypoint looks for the paths relative to itself? Do you think we could patch the code to look out of `bin`, so that scripts like "ssh" and "kill" doesn't end up in the users `$PATH`?

Reply:
Hmm, at the moment everything id done relative to `BASEDIR`, and that is why I put it all in the bin folder. So you think we can patch "${BASEDIR}" with something like `${nixpath-vmctl}/lib` in the scripts?

Reply:
ssh and kill will not be  in `$PATH` because it will in `$out/bin/cmd/ssh|kill`, so technically it should be fine?

Reply:
Adding a `../` would suffice, but you could override `BASEDIR` to be `${placeholder "out"}/` if you'd like.

Reply:
In the code, we get BASEDIR and use "${BASEDIR}" subsequently. I was thinking of using substituteInPlace to replace "${BASEDIR}" with ".." . I think dollar symbol is confusing nix. How do we get subsitituteInPlace to replace the whole `${BASEDIR}` with `..`? Probably a noob question.

Reply:
Ah, sorry. I didn't intend for you to replace the entirety of BASEDIR with `..`, it was more like "if BASEDIR always points to `$out/bin` and does so in a dynamic way, then maybe we can override `BASEDIR` to be `$BASEDIR/..` somehow".

As for the dollar symbol thing, nix uses `${}` to interpolate it's own variables. You can write `''${}` to escape the nix interpolation and leave it for bash.

Seeing as how `BASEDIR` is only set once, maybe something like this would work?

```
postPatch = ''
  substituteInPlace vmctl \
    --replace 'BASEDIR="$(dirname "$(readlink -f "''${BASH_SOURCE[0]}")")"' 'BASEDIR="${placeholder "out"}"'
'';
```

Reply:
Yes, that looks good to me. Thanks for all the comments :)